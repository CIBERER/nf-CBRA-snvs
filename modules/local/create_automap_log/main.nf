process CREATE_AUTOMAP_LOG {
    label 'process_single'
    
    conda "conda-forge::python=3.9.5 conda-forge::pandas=1.3.3"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/pandas:2.2.1':
        'biocontainers/pandas:2.2.1' }"

    input:
    val status_list

    output:
    path "automap_execution_log.txt", emit: log_file
    path "versions.yml", emit: versions

    script:
    """
    #!/usr/bin/env python3
    
    from datetime import datetime
    import subprocess
    
    # Parse the status list string
    status_data_str = '''${status_list}'''
    
    # Create log file
    with open("automap_execution_log.txt", "w") as log:
        log.write("AUTOMAP Execution Log\\n")
        log.write("=" * 50 + "\\n")
        log.write(f"Generated: {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}\\n")
        log.write("=" * 50 + "\\n\\n")
        
        try:
            # Remove outer brackets and split by '], ['
            entries = status_data_str.strip('[]').split('], [')
            
            for entry in entries:
                entry = entry.strip('[]')
                # Split by comma, but be careful with nested structures
                parts = entry.split(', ', 2)  # Split into max 3 parts
                
                if len(parts) >= 3:
                    meta = parts[0].strip()
                    status = parts[1].strip()
                    vcf = parts[2].strip()
                    
                    log.write(f"SAMPLE: {meta}\\n")
                    log.write(f"STATUS: {status}\\n")
                    log.write(f"VCF: {vcf}\\n\\n")
                    
        except Exception as e:
            log.write(f"Error parsing status list: {e}\\n")
            log.write(f"Raw data: {status_data_str}\\n")
        
        log.write(f"\\nLog generation completed at {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}\\n")

    # Create versions file using Python
    try:
        python_version = subprocess.check_output(['python3', '--version'], 
                                                stderr=subprocess.STDOUT, 
                                                text=True).strip().replace('Python ', '')
    except:
        python_version = "unknown"
    
    with open("versions.yml", "w") as versions:
        versions.write(f'"${task.process}":\\n')
        versions.write(f'    python: {python_version}\\n')
    """
}