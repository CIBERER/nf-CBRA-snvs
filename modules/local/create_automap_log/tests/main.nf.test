// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test CREATE_AUTOMAP_LOG
nextflow_process {

    name "Test Process CREATE_AUTOMAP_LOG"
    script "../main.nf"
    process "CREATE_AUTOMAP_LOG"

    tag "modules"
    tag "modules_nfcore"
    tag "CREATE_AUTOMAP_LOG"

    test("validate_automap - vcf") {

        when {
            process {
                """                
                input[0] = Channel.of([
                    [ id:'test'], // meta map
                    "PASS",
                    file("https://zenodo.org/records/15355673/files/Sample1_T1.gatk.PASS.vcf.gz?download=1")
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }

}
