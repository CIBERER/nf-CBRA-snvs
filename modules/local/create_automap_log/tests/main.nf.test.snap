{
    "validate_automap - vcf": {
        "content": [
            {
                "0": [
                    "automap_execution_log.txt:md5,73213781322a9239a17cfd974a610cd7"
                ],
                "1": [
                    "versions.yml:md5,5b0c36c698d75a118b031e73db6e16d6"
                ],
                "log_file": [
                    "automap_execution_log.txt:md5,73213781322a9239a17cfd974a610cd7"
                ],
                "versions": [
                    "versions.yml:md5,5b0c36c698d75a118b031e73db6e16d6"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-06-24T13:05:27.043072624"
    }
}